src/OrchardCore/OrchardCore.Contents.Core/AuthorizationServiceExtensions.cs

```diff
@@ -0,0 +1,69 @@
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using OrchardCore.Contents;
+using OrchardCore.Contents.Security;
+using OrchardCore.Security.Permissions;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Metadata.Models;
+
+namespace Microsoft.AspNetCore.Authorization
+{
+    public static class AuthorizationServiceExtensions
+    {
+        /// <summary>
+        /// Evaluate if we have a specific owner variation permission to at least one content type
+        /// </summary>
+        public static async Task<bool> IsAuthorizedToAContentTypeAsync(this IAuthorizationService service, ClaimsPrincipal user, Permission requiredPermission, IEnumerable<ContentTypeDefinition> contentTypeDefinitions)
+        {
+            foreach(var contentTypeDefinition in contentTypeDefinitions)
+            {
+                var permission = GetOwnerVariation(requiredPermission);
+                var contentTypePermission = ContentTypePermissionsHelper.ConvertToDynamicPermission(permission);
+                permission = ContentTypePermissionsHelper.CreateDynamicPermission(contentTypePermission, contentTypeDefinition);
+                
+                if(await service.AuthorizeAsync(user, permission, new ContentItem{ ContentType = contentTypeDefinition.Name, Owner = user.Identity.Name }))
```

Comment:
@jtkech @deanmarcussen The only thing remaining is to define if I should be injecting the ContentManager to create the fake ContentItem in this method. I personnally think that using the ContentManager everywhere is not a good solution everytime but I'd like to understand better your opinions about this.

Open to everyone comments.

Reply:
@Skrypt @deanmarcussen 

I'm fine here to use `new ContentItem` in place of `NewAsync()`

Hmm, maybe unless if an auth handler check on the contentItem resource other things than the type or the owner, things that would be updated / added by `NewAsync()` e.g. by an activating / activated handler.

Just now i'm thinking about an auth handler that would check if a content item has a given part / field, but the handler could inject the definition manager, and i don't think that we will support this kind of permission ;)

So because these other things would be coupled to the content type, the only parameter that we pass to `NewAsync()`, when `NewAsync()` is only used for permissions purposes, I'm fine to rather use `new ContentItem()`. And if all agree we could do the same in the other places where `NewAsync()` is only used for the same reason.

Reply:
Yes, and without saying it too loud I think this might cause eventual security issues. Anyone could take the contentItem passed in the activating / activated handlers and modify intentionally the Owner == "Admin" so that he can bypass the actual security. Here is why I wanted also to use a strong identifier for user accounts ... see UserId PR comments. 😉

Reply:
Ok so for me, it's actually what @jtkech said about Parts and extensibility.

I looked back at some old O1 security code I had for managing User Groups

The code is not exactly the same, O1 had `IAuthorizationServiceEventHandler` but it's got good similarities to an `AuthorizationHandler` and the code should transplant over if/when I need to.

The first thing the custom  `IAuthorizationServiceEventHandler` does is 
```
var part = context.Content.As<UserGroupPart>();
```

so by not calling `.NewAsync()` everywhere the `ContentItem` parts are not loaded onto the kernel, and there is a loss of extensibility.

Now to be fair, the O1 code I looked at wasn't really interested in whether or not to show the `New` content item menu. And technically it could look at the content definition as @jtkech says. But that's ...

- more complicated
- only necesary because here you are making a breaking change to the concept of never creating a `ContentItem` directly, and bypassing it the content manager. Which just feels like a bad / unnecessary modification.

Regards your security concerns @Skrypt I would keep it for exactly that reason. It allows a developer to write extensible code, that can attach different permissions. Think about User Grouping, and flexing ownership of content through the use of groups.

A developer will always be able to read content items that aren't owned by them. A simple query in a view through `ISession` would do it, and we've always said that security is about providing permissions to secure what you can do through the ui. Not preventing malicious developers.

Reply:
Yeah, I see your point. Though we can create specific AuthorizationHandlers for each use cases now too. And we can call them by order of priority. Here, when we create a ContentTypeAuthorizationHandler, I believe it is for handling permissions related with ContentTypes. There is also nothing preventing creating a ContentPartAuthorizationHandler specific for the need of when using a UserGroupPart.

So, here if I would use the contentManager.NewAsync() method it would weld the ContentParts to the content item. But, it is not here that I would require to weld them but in the Controller or View that requires to render it. So, in this AuthorizationHandler, the only thing I should be authorizing is related with "ContentType" only.

Reply:
I agree that bypassing the contentManager here in that case is not a necessity. Though, doing so, prevents from adding extra processing that would happen when processing those event handlers. So, over time it would prevent from adding extra processing time every time we need to evaluate a permission. If we can do so then I think it makes sense. This is why mainly I'm asking everyone to think about if it's a necessity here to use the ContentManager.NewAsync() in that specific place. Else, easier for me to just use it and merge the PR... 😉

Reply:
> Else, it would be easier for me to just use ContentManager.NewAsync() and merge the PR..

Please just do this :) 

Don't kill extensibility... even if you can't see a use for it someone will need it one day.
This is not on the front end, so the processing cost of running a handler is not something to be super concerned about (IMHO)

Reply:
I agree to disagree with your opinion on that one. Because it doesn't kill the extensibility when we Authorize a content item on a admin controller action we use the ContentManager. We use it when it's needed. But I will just do this because we're all tired of this PR 😄

Reply:
Done, now IsAuthorizedToAContentTypeAsync is dependend on the IContentManager. Which means it will create a new contentItem for each ContentType present in the ContentDefinitions and it will trigger the handlers for each of those content types. So basically it will exponentially become slower if someone does something funky in it's ActivatedAsync/ActivatingAsync,  InitializingAsync/InitializedAsync events.